storage/remote/otlptranslator/prometheusremotewrite/helper.go

```diff
@@ -521,51 +449,50 @@ func (c *PrometheusConverter) addSummaryDataPoints(ctx context.Context, dataPoin
 // createLabels returns a copy of baseLabels, adding to it the pair model.MetricNameLabel=name.
 // If extras are provided, corresponding label pairs are also added to the returned slice.
 // If extras is uneven length, the last (unpaired) extra will be ignored.
-func createLabels(name string, baseLabels []prompb.Label, extras ...string) []prompb.Label {
-	extraLabelCount := len(extras) / 2
-	labels := make([]prompb.Label, len(baseLabels), len(baseLabels)+extraLabelCount+1) // +1 for name
-	copy(labels, baseLabels)
+func (c *PrometheusConverter) createLabels(name string, baseLabels labels.Labels, extras ...string) labels.Labels {
```

Comment:
At this point this function is more a `addLabels`, no? See my other comment around passing builders around for this.